source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,169 @@
+(nixos-vms)=
+
+# NixOS virtual machines
+
+One of the most important features of NixOS is the ability to configure the entire system declaratively, including packages to be installed, services to be run, as well as other settings and options.
+
+NixOS configurations can be used to test and use NixOS using a virtual machine, independently of an installation on "bare metal" or your laptop.
+
+A NixOS configuration is a Nix language function following the [NixOS module](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules) convention. 
+
+## What will you learn?
+
+This tutorial serves as an introduction creating virtual NixOS machines using Nix.
+Virtual machines are a practical tool for debugging NixOS configurations.
+
+## What do you need?
+
+- A working installation of [Nix Package Manager](https://nixos.org/manual/nix/stable/installation/installation.html) or [NixOS](https://nixos.org/manual/nixos/stable/index.html#sec-installation).
+- Basic knowledge of the [Nix language](https://nixos.org/manual/nix/stable/language/index.html).
+
+## Starting from the default NixOS configuration
+
+On NixOS, by default, the configuration file is located at `/etc/nixos/configuration.nix`.
+For CLI commands that use a configuration, its file path can be specified.
+In this tutorial we use this strategy, because the goal is not to install or update the current NixOS, but to debug a specific configuration.
+
+On NixOS, you can use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions.[^nixosconf]
+You can create a NixOS configuration in your working directory:
+```shell-session
+nixos-generate-config --dir ./
+```
+
+In the working directory you will find two files: `configuration.nix` and `hardware-configuration.nix`.
+
+The `hardware-configuration.nix` file is specific to the hardware the `nix-generate-config` is being run on.
+We can ignore that file for this tutorial because it has no effect inside a virtual machine.
+
+The `configuration.nix` file contains various suggestions for the initial setup of a desktop computer.
+Without comments the configuration file contains the following content:
+```nix
+{ config, pkgs, ... }:
+{
+  imports =
+    [
+      ./hardware-configuration.nix
+    ];
```

Comment:
Not a mandatory change, because it's not entirely true to your statement of the file without comments, but it makes it easier to describe the changes down the line.
```suggestion
  imports =  [ ./hardware-configuration.nix ];
```